miden-crypto/src/merkle/smt/large/mod.rs

```diff
@@ -194,7 +194,10 @@ type Leaves = super::Leaves<SmtLeaf>;
 #[derive(Debug)]
 pub struct LargeSmt<S: SmtStorage> {
     storage: S,
-    in_memory_nodes: Vec<Option<Box<InnerNode>>>,
+    /// Flat vector representation of in-memory nodes.
+    /// Index 0 is unused; index 1 is root.
+    /// For node at index i: left child at 2*i, right child at 2*i+1.
+    in_memory_nodes: Vec<Word>,
```

Comment:
Consider adding tests to verify the flat Vec<Word> layout correctly maps tree structure. Check index 0 unused, root at index 1, left/right children at 2*i/2*i+1. Ensure NUM_IN_MEMORY_NODES capacity is pre-allocated without re-allocations.